torchrl/envs/transforms/transforms.py

```diff
@@ -2433,3 +2433,51 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(decay={self.decay:4.4f},"
             f"eps={self.eps:4.4f}, keys={self.in_keys})"
         )
+
+
+class RewardSum(Transform):
+    """Tracks the accumulated reward of each episode.
+
+    This transform requires ´reward´ to be input key. If that is not the case,
+    the transform has no effect.
+    """
+
+    inplace = True
+
+    def __init__(
+        self,
+        in_keys: Optional[Sequence[str]] = None,
+        out_keys: Optional[Sequence[str]] = None,
+    ):
+        in_keys = ["reward"]
+        if out_keys is None:
+            out_keys = ["episode_reward"]
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Resets episode rewards."""
+        if "reset_workers" in tensordict.keys():
```

Comment:
what happens if we apply reset to an env that is not batched? I don't see the `"epsode_reward"` being reset in that case

Reply:
Modified the code to include the non-batched case